### PR TITLE
RBAC integration

### PR DESCRIPTION
**Pull request by JumasJM**
Type: Improvement
JIRA: DEP-686

### CODE REVIEW

File: archive/src/main/java/io/orkes/conductor/dao/postgres/archive/PostgresArchiveDAO.java [hunk 1]
```diff
@@ -536,4 +540,124 @@ public List<WorkflowModel> getWorkflowFamily(String workflowId, boolean summaryO
...
+
+        }
+        return new String[0];
+    }
+
+    @Override
+    public List<String> getUserWorkflowIds(List<String> groupsAndRoles) {
+        final String GET_USER_WORKFLOWS = "SELECT " +
+                "workflow_id FROM archive.workflow_archive " +
+                "WHERE rbac_labels && ARRAY[?]::text[]";
```
Comment (Vasyl-9) [inline archive/src/main/java/io/orkes/conductor/dao/postgres/archive/PostgresArchiveDAO.java]: If you use ';' in the end of SQL command in another query then add it also here.

File: persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresMetadataDAO.java [hunk 2]
```diff
@@ -575,4 +575,120 @@ private String insertOrUpdateTaskDef(TaskDef taskDef) {
...
+    }
+
+    @Override
+    public List<WorkflowDef> getUserWorkflowDefs(List<String> groupsAndRoles) {
+        final String GET_USER_WORKFLOWS = "SELECT json_data " +
+                "FROM meta_workflow_def " +
+                "WHERE EXISTS (" +
+                "SELECT 1 " +
+                "FROM json_array_elements_text((json_data::json ->>'description')::json ->'labels') AS label " +
+                "WHERE label = ANY(?))";
```
Comment (Vasyl-9) [inline persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresMetadataDAO.java]: Also add ';' in the end of SQL command.

File: persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresMetadataDAO.java [hunk 3]
```diff
@@ -575,4 +575,120 @@ private String insertOrUpdateTaskDef(TaskDef taskDef) {
...
+    }
+
+    @Override
+    public List<TaskDef> getUserTaskDefs(List<String> groupsAndRoles) {
+        final String GET_USER_TASKS = "SELECT json_data " +
+                "FROM meta_task_def " +
+                "WHERE EXISTS (" +
+                "SELECT 1 " +
+                "FROM json_array_elements_text((json_data::json ->>'description')::json ->'labels') AS label " +
+                "WHERE label = ANY(?))";
```
Comment (Vasyl-9) [inline persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresMetadataDAO.java]: the same as previous comment

**Review by Vasyl-9 — CHANGES_REQUESTED** [review]

File: archive/src/main/java/io/orkes/conductor/dao/postgres/archive/PostgresArchiveDAO.java (hunk 1 above)
Comment (JumasJM) [reply archive/src/main/java/io/orkes/conductor/dao/postgres/archive/PostgresArchiveDAO.java]: added

File: persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresMetadataDAO.java (hunk 2 above)
Comment (JumasJM) [reply persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresMetadataDAO.java]: added

File: persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresMetadataDAO.java (hunk 3 above)
Comment (JumasJM) [reply persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresMetadataDAO.java]: added

File: persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresIndexDAO.java [hunk 4]
```diff
@@ -299,4 +299,9 @@ public long getWorkflowCount(String query, String freeText) {
         logger.info("getWorkflowCount is not supported for postgres indexing");
         return 0;
     }
+
+    @Override
+    public SearchResult<WorkflowSummary> getSummaries(SearchResult<String> searchResultIds) {
+        throw new UnsupportedOperationException("Not implemented yet.");
+    }
```
Comment (Vasyl-9) [inline persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresIndexDAO.java]: Based on previous methods I will be standardize this class to one form (fail-fast with UnsupportedOperationException or not fail-fast with logging error and returning some negative value - 0, null or false)

**Review by Vasyl-9 — CHANGES_REQUESTED** [review]

File: persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresIndexDAO.java (hunk 4 above)
Comment (JumasJM) [reply persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresIndexDAO.java]: Resolved.

**Review by Vasyl-9 — APPROVED** [review]

**Review by Jozefiel — APPROVED** [review]

**Pull request merged**